### ISSUE TITLE
Subscribe, unsubscribe, resubscribe

### ISSUE THREAD

**Issue by dionjwa** [denodrivers/redis#415]
I am getting odd errors when my server:

 - creates a subscription to a specific key `subscribeClient.subscribe(key).then(async (subscription:any) => {`
 - listens for messages
 - breaks out of the `for await (const { channel, message } of subscription.receive()) {` loop
 - ? do I unsubscribe here? It doesn't seem to help
 - subscribes *again* to the same key
   - this subscription usually doesn't return

There is something odd going on with `subscribe` and `unsubscribe` (or just exiting the subscription `for await` loop)

What is the correct expected pattern to `subscribe`->`unsubscribe`->`subscribe` again to the same key?

Thank you!

**Comment by dionjwa** [comment]
It might be related to https://github.com/denodrivers/redis/issues/390

**Comment by uki00a** [comment]
Thanks for reporting the issue @dionjwa!

> ? do I unsubscribe here? It doesn't seem to help

No, it is not unsubscribed if it just exits the loop.

> What is the correct expected pattern to `subscribe`->`unsubscribe`->`subscribe` again to the same key?

The following code seemed to work properly. If possible, could you share the code that is causing the problem? I would like to look into it.

```javascript
import { connect } from "./redis.ts";
import { assertEquals } from "./vendor/https/deno.land/std/assert/mod.ts";

const opts = {
  hostname: "127.0.0.1",
  port: 6379
};
const redis = await connect(opts);
const pub = await connect(opts);
const channel = "foobar";
const sub = await redis.subscribe(channel);
const messages = [];
try {
  const promise = (async () => {
    for await (const m of sub.receive()) {
      messages.push(m);
      break;
    }
  })();
  await pub.publish(channel, "message");

  await promise;
  await sub.unsubscribe(channel);
  await sub.subscribe(channel);

  const promise2 = (async () => {
    for await (const m of sub.receive()) {
      messages.push(m);
      break;
    }
  })();
  await pub.publish(channel, "message 2");

  await promise2;
} finally {
  sub.close();
  redis.close();
  pub.close();
}
assertEquals(messages, [
  { channel, message: "message" },
  { channel, message: "message 2" }
]);
```

**Comment by uki00a** [comment]
P.S. I believe #390 has been resolved in [v0.32.1](https://github.com/denodrivers/redis/releases/tag/v0.32.1). Maybe this problem has also been resolved.

**Comment by dionjwa** [comment]
Yes, I am not noticing this issue anymore, this issue can be closed.
